contrib/feeds/proofpoint/proofpoint.py

```diff
@@ -313,9 +312,31 @@ def _make_threat_nodes(threat, context, tags):
         # if they do, do not parse the threat a second time ?
         threat_nodes = []
         if 'url' in threats:
-            threat_nodes.append(
-                Url.get_or_create(
-                    value=threats['url']['threat'], context=[context]))
+            #Proofpoint sometimes supplies a hostname marked as a Url.
+            #validation rejected such hostnames
+            #this trys a Url or a Hostname and appends it as the proper class/type
+            try:
```

Comment:
Couldn't you use `Observable.add_text()` (see [docs](http://yeti-platform.readthedocs.io/en/latest/objects.html#core.observables.Observable.add_text)) instead of trying first with hostnames and then with URLs? The method will do the guessing for you.